zstacklib/zstacklib/utils/gpu.py

```
@@ -657,34 +661,62 @@ def check_huawei_npu_is_isolated(npu_id, all_npu_ids, iswindows=False):
 
 
+def _check_npu_isolation_by_topo(npu_id, iswindows=False):
+    """
+    Fallback isolation detection via topo matrix.
+    An isolated NPU has zero HCCS connections (all links show SYS or PHB).
+    """
+    cmd = "npu-smi info -t topo -i {0}".format(npu_id)
+    if iswindows:
+        cmd = cmd.replace(" ", "|")
+
+    r, o, e = bash_roe(cmd)
```

Ruff (0.15.2)

[error] `bash_roe` may be undefined, or defined from star imports

(F405)